src/util/expr_util.cpp

```diff
@@ -19,6 +19,19 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "namespace.h"
 #include "arith_tools.h"
 
+bool is_lvalue(const exprt &expr)
+{
+  if(expr.id()==ID_index)
+    return is_lvalue(to_index_expr(expr).op0());
```

Comment:
No point in casting and then using `exprt`'s `op0` method -- use `array()` and `compound()` for `index` and `member_exprt` respectively or don't cast.

Reply:
Resolved in latest version.